homeassistant/components/risco/alarm_control_panel.py

```diff
@@ -108,6 +107,14 @@ def __init__(
             self._attr_supported_features |= STATES_TO_SUPPORTED_FEATURES[state]
 
     @property
+    def code_format(self) -> CodeFormat | None:
+        """Code format or None if no code is required."""
+        return (
+            CodeFormat.NUMBER
+            if self._attr_code_arm_required or self._code_disarm_required
```

Comment:
These values don't change over time, so we can set `self._attr_code_format` within `__init__()`

Reply:
It's a matter of taste. If you take a look at the parent class *AlarmControlPanelEntity* you can see that it has a *code_format()* function that returns *self._attr_code_format*, so having a function just overrides the parent.

This brings me to your second point, re. tests. I had a look at your test code and realized it's going to take me many hours, if not days, to figure out exactly what it's doing, especially considering that I have no experience with pytest and can't get a debugger to work properly on it. Therefor, and assuming it's going to take you just minutes to do, I suggest you take my single line of code and add it yourself along with any tests you deem appropriate. 

By the way, it's a goo opportunity to thank you for the awesome work on this integration.

Reply:
> It's a matter of taste. If you take a look at the parent class AlarmControlPanelEntity you can see that it has a code_format() function that returns self._attr_code_format, so having a function just overrides the parent.

It could be a matter of taste, but that's the preferred way to handle it in HA. The parent class has this implementation exactly for cases like these.

As for testing, I'll try to get to it soon, but it might take me a while. In the meantime, you might want to take a look at the tests that end with `sets_with_incorrect_code` as an example.